backend/src/test/java/quarano/actions/quarantine/QuarantineEndCheckerTest.java

```diff
@@ -0,0 +1,164 @@
+			new TrackedPerson("Olaf", "Beispiel"),
+			CaseType.INDEX,
+			DEPARTMENT
+	).setQuarantine(QUARANTINE);
+
+	private static final TrackedCase TRACKED_CASE_D = new TrackedCase(
+			new TrackedPerson("Elfride", "Exampel"),
+			CaseType.CONTACT,
+			DEPARTMENT
+	);
+	private static final Streamable<TrackedCase> TRACKED_CASES =
+			Streamable.of(TRACKED_CASE_A, TRACKED_CASE_B, TRACKED_CASE_C, TRACKED_CASE_D);
+	private static final int ONCE = 1;
+	private static final int NEVER = 0;
+	private static final int THREE = 3;
+
+	@Mock private TrackedCaseRepository cases;
+	@Mock private ActionItemRepository items;
+
+	private QuarantineEndChecker quarantineEndChecker;
+
+	@BeforeEach
+	void setUp() {
+		quarantineEndChecker = new QuarantineEndChecker(cases, items);
+		when(cases.findAll()).thenReturn(TRACKED_CASES);
+	}
+
+	@Test
+	void expectToCreateThreeQuarantineEndingActionItem() {
+		prepareTestDateTime(DATE_2020_05_17);
+		prepareNonExistingActionItems();
+
+		quarantineEndChecker.checkEndingQuarantinesPeriodically();
+
+		ArgumentCaptor<QuarantineEndingActionItem> captor = verifyRepositorySaveInteraction(THREE);
+		verifyCreatedQuarantineEndingActionItems(captor);
+		verifyNonExistenceOfSuitableActionItems();
+	}
+
+	@Test
+	void expectToCreateNoQuarantineEndingActionItemsDueToDateFiltering() {
+		prepareTestDateTime(DATE_2020_05_10);
+
+		quarantineEndChecker.checkEndingQuarantinesPeriodically();
+
+		ArgumentCaptor<QuarantineEndingActionItem> captor = verifyRepositorySaveInteraction(NEVER);
+		verifyCreatedQuarantineEndingActionItems(captor);
+	}
+
+	@Test
+	void expectToCreateOneQuarantineEndingActionItemDueToExistingActionItems() {
+		prepareTestDateTime(DATE_2020_05_17);
+		prepareTwoExistingActionItems();
+
+		quarantineEndChecker.checkEndingQuarantinesPeriodically();
+
+		ArgumentCaptor<QuarantineEndingActionItem> captor = verifyRepositorySaveInteraction(ONCE);
+		verifyCreatedQuarantineEndingActionItems(captor);
+		verifyQuarantineEndingActionItemForTrackedCaseC(captor);
+	}
+
+	private void prepareTwoExistingActionItems() {
+		when(items.findQuarantineEndingActionItemsFor(TRACKED_CASE_A.getTrackedPerson().getId()))
+				.thenReturn(ActionItems.of(new QuarantineEndingActionItem()));
+		when(items.findQuarantineEndingActionItemsFor(TRACKED_CASE_B.getTrackedPerson().getId()))
+				.thenReturn(ActionItems.of(new QuarantineEndingActionItem()));
+		when(items.findQuarantineEndingActionItemsFor(TRACKED_CASE_C.getTrackedPerson().getId()))
+				.thenReturn(ActionItems.empty());
+	}
+
+	private void verifyQuarantineEndingActionItemForTrackedCaseC(
+			ArgumentCaptor<QuarantineEndingActionItem> captor) {
+		assertThat(captor.getValue())
+				.isEqualTo(new QuarantineEndingActionItem(TRACKED_CASE_C.getTrackedPerson().getId()));
+	}
+
+	private void prepareTestDateTime(LocalDate date20200510) {
+		DateTimeUtils.setCurrentMillisFixed(date20200510.toEpochSecond(LocalTime.MIN, ZoneOffset.UTC));
```

Comment:
What do you think about this testable approach? I have some good experience with `jodatime`. 👍 🚀